core/server/common/src/main/java/alluxio/master/Master.java

```diff
@@ -24,7 +22,7 @@
 /**
  * This interface contains common operations for all masters.
  */
-public interface Master extends JournalCheckpointStreamable {
+public interface Master extends Iterable<JournalEntry> {
```

Comment:
Master can be considered as a collection of JournalEntries:) The more important reason for me to use Iterable is to make use some util functions such as Iterators::concat to concatenate multiple iterators. Without that, i need to write those utils by myself.

@jsimsa @gpang  how do you guys think?